server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/blobstore/BlobStoreModulesChooser.java

```diff
@@ -37,20 +39,16 @@
 import com.google.inject.name.Names;
 
 public class BlobStoreModulesChooser {
-    static class CassandraDeclarationModule extends AbstractModule {
+    static class CassandraDumbBlobStoreDeclarationModule extends AbstractModule {
         @Override
         protected void configure() {
             install(new CassandraBlobStoreDependenciesModule());
 
             bind(DumbBlobStore.class).to(CassandraDumbBlobStore.class);
-
-            bind(BlobStore.class)
-                .annotatedWith(Names.named(CachedBlobStore.BACKEND))
-                .to(DeDuplicationBlobStore.class);
         }
     }
 
-    static class ObjectStorageDeclarationModule extends AbstractModule {
+    static class ObjectStorageDumdBlobStoreDeclarationModule extends AbstractModule {
         @Override
         protected void configure() {
             install(new ObjectStorageDependenciesModule());
```

Comment:
I tried to look a bit into the Guice errors... The problem might possibly coming from here I think?

You need to drop your bind below as it conflicts with what you declared in `chooseStoragePolicyModule` and here I would say you need to bind the  `DumbBlobStore` to a... `ObjectStorageDumdBlobStore` that still not exist it seems (in an other PR? I forgot)

Reply:
Yes, that class is introduced by the S3 PR, and this work is conditioned to it.

Reply:
rebased on #3614 the NoopDumbBlobStore should fix the guice issue for the moment